### PR TITLE
Including OtherLegendDataSource to config and templates

### PR DESCRIPTION
**Pull request by voisardf**
Adds the mandatory 'Other legend' to the PDF extract.
Needs to be completed in the mapfish_print proxy script

- [x] Processing OtherLegend from 
```
'RestrictionOnLandownership': [{
        'Map': {
            'OtherLegend': [{
                'Theme': {
                    'Text': {
                        'Text': "Plan d'affectation",
                        'Language': 'fr'
                    },
                    'Code': 'LandUsePlans'
                },
                'TypeCodelist': u 'https://sitnssl.ne.ch/ogc-pyramid-oereb/wms?SERVICE=WMS&REQUEST=GetStyles&VERSION=1.1.1&LAYERS=r73_affectations_primaires',
                'LegendText': [{
                    'Text': u 'zone industrielle',
                    'Language': 'fr'
                }],
                'SymbolRef': 'http://localhost:6543/oereb/image/symbol/LandUsePlans?TEXT=eyJmciI6ICJ6b25lIGluZHVzdHJpZWxsZSJ9&CODE=1205',
            }],
        }
    }],
```
- [x] Add OtherLegendDataSource to config.yml
- [x] Add OtherLegendSupreport to topicpage.jrxml

### CODE REVIEW

**Comment by voisardf** [comment]
Ok, I also added the python part in mapfish_print. @sbrunner please give it an other glance, but I have results :)
[20171110060728_extract.pdf](https://github.com/camptocamp/pyramid_oereb/files/1462103/20171110060728_extract.pdf)
[20171110060233_extract.pdf](https://github.com/camptocamp/pyramid_oereb/files/1462104/20171110060233_extract.pdf)
[20171110060826_extract.pdf](https://github.com/camptocamp/pyramid_oereb/files/1462105/20171110060826_extract.pdf)

the symbols missing are just not in the database yet 😉 
I guess I earned my right for a weekend-beer...

**Comment by voisardf** [comment]
now it should be ready to merge :)

File: pyramid_oereb/contrib/print_proxy/mapfish_print.py [hunk 1]
```diff
@@ -234,7 +249,7 @@ def __call__(self, value, system):
 
         extract_dict['RealEstate_RestrictionOnLandownership'] = restrictions
         # End one restriction entry per theme
-
+
```
Comment (sbrunner) [inline pyramid_oereb/contrib/print_proxy/mapfish_print.py]: :-/

File: pyramid_oereb/contrib/print_proxy/mapfish_print.py [hunk 2]
```diff
@@ -117,6 +118,20 @@ def __call__(self, value, system):
             }
             restriction_on_landownership['legend'] = restriction_on_landownership['Map'].get(
                 'LegendAtWeb', '')
+
+            # Legend of  other visible restriction objects in the topic map
+            restriction_on_landownership['OtherLegend'] = restriction_on_landownership['Map'].get('OtherLegend','')
+            if restriction_on_landownership['OtherLegend'] != '':
```
Comment (sbrunner) [inline pyramid_oereb/contrib/print_proxy/mapfish_print.py]: `OtherLegend` is a array of dict, not a string => 
```
restriction_on_landownership['OtherLegend'] = restriction_on_landownership['Map'].get('OtherLegend', [])
```
and remove the if ...

File: pyramid_oereb/contrib/print_proxy/mapfish_print.py [hunk 3]
```diff
@@ -117,6 +118,20 @@ def __call__(self, value, system):
...
+                for legend_item in restriction_on_landownership['OtherLegend']:
+                    self._multilingual_text(legend_item, 'LegendText')
+
+            for legend_entry in restriction_on_landownership['OtherLegend']:
+                if legend_entry != '':
```
Comment (sbrunner) [inline pyramid_oereb/contrib/print_proxy/mapfish_print.py]: Is it relay needed? (comparing a dict with a string ...)

File: pyramid_oereb/contrib/print_proxy/mapfish_print.py [hunk 4]
```diff
@@ -117,6 +118,20 @@ def __call__(self, value, system):
...
+                    other_legend = []
```
Comment (sbrunner) [inline pyramid_oereb/contrib/print_proxy/mapfish_print.py]: Not used ...

File: pyramid_oereb/contrib/print_proxy/mapfish_print.py [hunk 5]
```diff
@@ -117,6 +118,20 @@ def __call__(self, value, system):
...
+                    for element in legend_entry.keys():
+                        if not element in ['LegendText', 'SymbolRef', 'TypeCode']:
+                            del legend_entry[element]
```
Comment (sbrunner) [inline pyramid_oereb/contrib/print_proxy/mapfish_print.py]: better: 
```
for element in ['LegendText', 'SymbolRef', 'TypeCode']:
  if element in legend_entry:
    del legend_entry[element]
```

**Comment by voisardf** [comment]
Merging if it's ok for you now?

File: pyramid_oereb/contrib/print_proxy/mapfish_print.py [hunk 6]
```diff
@@ -117,6 +117,19 @@ def __call__(self, value, system):
             }
             restriction_on_landownership['legend'] = restriction_on_landownership['Map'].get(
                 'LegendAtWeb', '')
+
+            # Legend of  other visible restriction objects in the topic map
+            restriction_on_landownership['OtherLegend'] = restriction_on_landownership['Map'].get(
+                'OtherLegend', [])
+            if restriction_on_landownership['OtherLegend'] != '':
```
Comment (sbrunner) [inline pyramid_oereb/contrib/print_proxy/mapfish_print.py]: Does it happens? (to be false)

Comment (voisardf) [reply pyramid_oereb/contrib/print_proxy/mapfish_print.py]: Forgot that one, sorry. 🙈 
It just happens to be None sometimes

**Pull request merged**